### PR TITLE
[XPU] add support for local path compilation

### PR DESCRIPTION
**Pull request by skywalker2012**
### PR types

Others

### PR changes

Others

### Description

This PR add support for XPU+Paddle local path compilation, can be used as a more flexible way under some situations. 
For better experience, please see below example of usage:

```
cmake .. -DPY_VERSION=3.9 \
    -DCMAKE_BUILD_TYPE=Release \
    -DWITH_GPU=OFF \
    -DWITH_XPU=ON \
    -DON_INFER=ON \
    -DWITH_PYTHON=ON \
    -DWITH_MKL=ON \
    -DWITH_XPU_BKCL=ON \
    -DWITH_TESTING=ON \
    -DWITH_DISTRIBUTE=ON \
    -DWITH_XPU_XHPC=ON \
    -DXPU_BASE_URL=/workspace/paddle_develop/Paddle/output/xpu_portfolio_ubuntu/toolkit/modules

```

**_XPU_BASE_URL_** is treated as a local absolute path in this way, which should contain necessary modules folder like below:
- xre-ubuntu_x86_64/
- xhpc-ubuntu_x86_64/
- xccl_socket-ubuntu_x86_64/

### CODE REVIEW

**Review by runzhech — APPROVED** [review]
LGTM

**Review by lj970926 — APPROVED** [review]
LGTM

File: tools/xpu/pack_paddle_depence.sh [hunk 1]
```diff
@@ -32,41 +26,108 @@ XDNN_DIR_NAME=$4
...
+    cp -r ${LOCAL_PATH}/${XHPC_DIR_NAME}/xdnn/include/* xpu/include/
+    cp -r ${LOCAL_PATH}/${XHPC_DIR_NAME}/xdnn/so/* xpu/lib
+
+    cp -r ${LOCAL_PATH}/${XHPC_DIR_NAME}/xfa/include/* xpu/include/xhpc/xfa
+    cp -r ${LOCAL_PATH}/${XHPC_DIR_NAME}/xfa/so/* xpu/lib/
+}
+
+if [[ $XRE_URL != *"$BOS_PATTERN"* ]]; then
+    # below is local way
+    build_from="local"
```
Comment (houj04) [inline tools/xpu/pack_paddle_depence.sh:117]: 这里的`build_from`变量好像只有定义，没见哪里有使用？

Comment (skywalker2012) [reply tools/xpu/pack_paddle_depence.sh:117]: build_from

是的，主要是为了让编译的人看到目前走的编译方式，相当于echo

**Review by houj04 — APPROVED** [review]

**Review by HarperCy — APPROVED** [review]

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**